lib/diagnostic.rb

```diff
@@ -13,6 +15,7 @@ def longest(enum)
   # enumerable supplied (i.e. apply the `+` operator)
   # Assume each entry implements a `+` method
   def sum(enum)
+    enum.reduce(0) { |m, o| m += m}
```

Comment:
Correct. This could also be written more simply as `enum.reduce(&:+)` (or just `enum.reduce(:+)` in Ruby 2.0 and above).